extensions/vault/runtime/src/main/java/io/quarkus/vault/transit/Plaintext.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.vault.transit;
+
+import static io.quarkus.vault.runtime.StringHelper.bytesToString;
+import static io.quarkus.vault.runtime.StringHelper.stringToBytes;
+
+import java.util.Arrays;
+
+public class Plaintext {
```

Comment:
@vsevel May be it should be renamed ? I thought, when I looked at the API, it was a String wrapper. It actually has simplified the API, this class :-), but since it can hold bytes not only from String. How about `ByteContent` ?

Reply:
I know it looks a bit strange, but this maps directly to the [plaintext](https://www.vaultproject.io/api/secret/transit/index.html#plaintext) attribute (which is a base64 encoded array of bytes) from the vault api, and spring vault did the same [class Plaintext](https://docs.spring.io/spring-vault/docs/current/api/org/springframework/vault/support/Plaintext.html)
I feel we should stay close to the api.

Reply:
@vsevel IMHO if we can do something differently from Spring API then lets do it.
It looks like Spring API did not get it right, because the [plaintext](https://www.vaultproject.io/api/secret/transit/index.html#plaintext) is indeed a plaintext because it is base64 encoded, while what the API has with `Plaintext` in our case is just `byte[]` yet to be base-64 encoded (and it is very good because asking users to base64 encode first would be an extra problem for them). I'm sorry if I'm biased but if I saw `BinaryContent` I would be less confused :-).

Reply:
I changed it to `ClearData`. Then to be consistent I changed the `Ciphertext` to `EncryptedData`.
But that is one of those things where I think that when you are working with vault, you should be reading vault's documentation. And then things like `plaintext` should be clear from that, and keeping the names in synch between the quarkus extension and vault's api makes things easier to understand. I do not think we should reinvent concepts. We should keep the spirit of the original api. 
Part of your confusion could be related actually to the fact that you may not have read vault's documentation. And I could argue that somebody who is aware, will probably be confused by the fact that we called it `ClearData` in quarkus. So I will have to add a comment in the javadoc saying that `ClearData` maps actually to the `plaintext` attribute.
Let's see what others think.

Reply:
@vsevel well, as you saw I was not asking for `ClearData` either. I'm sorry but I'm not following your point. From the Vault docs about the `plaintext`:
```
Specifies base64 encoded plaintext to be encoded
``` 
It says clearly it is a base-64 encoded String.
You had `Plaintext` holding `byte[]` and not a base64 string. So I'm not sure why it would be confusing to anyone seeing a `byte[]` wrapper qualified as `BinaryContent`.
And if I get it right, it is not mapped to 'plaintext' 1:1, but its content is base64 encoded first. Furthermore why would users need to know about the low levels part of the Vault API at the Quarkus Vault API level ?

But how about a small compromise between `ClearData` and `BinaryContent` -> `BinaryData`, with the docs attached to it this data will be Base64 encoded before it is made available to Vault.

What do you think ?

Reply:
If you prefer, please rename back to Plaintext, I'm not insisting there must be some renaming, IMHO Plaintext was not adequately representing its content, and the fact some other API like Spring Vault chose to name a `byte[]` content as PlianText was not something I'd personally be concerned about, but please make a final decision about the name as you see fit

Reply:
Looking at the API, `ClearData` is actually looking not bad at all to me :-), it is different to `PlainText` but IMHO it is definitely `clearer` :-) what it is about. I'm not sure, `BinaryData` would probably be OK too. Up to you how to resolve it :-)

Reply:
Hello @sberyozkin 
Thanks for your feedback. I think part of the argument revolves around whether the quarkus extension should be:
 - a thin wrapper around an api's product, keeping a high level of fidelity
 - or its own abstraction with usage simplifications and value added, even if that means going away from the underlying product, which could be seen as an implementation detail in the extreme case.

Obviously I approached the problem from the first angle (and `Plaintext` in that context makes more sense because people can immediately relate to it in the documentation, even if I do agree may be the naming could have been better in vault's api), and it seems you are more leaning toward the second one (and `ClearData` or `BinaryContent/Data` look a lot better because in that case you want to favor your api's readability, and not fidelity). None of them are wrong. It is just 2 different perspectives. My comment was a bit harsh. Sorry for that.

As you can see I am torn between those 2 different perspectives. It could have been good to have others feedback - we need an odd number for a majority ;-) 
Now let's go with the more readable version.
I think vault's api went with `plain` to convey the fact that it is unencrypted, and `cipher` being its symmetric. I felt that if we were going for new names, then we should keep the symmetry and express that one type represents unencrypted data, and the other one encrypted data. That is the reason I was not comfortable with `BinaryContent/Data` as it does not make it obvious that it contains unencrypted data, plus I thought we should rename the encrypted one as well. So I propose we stay with `ClearData` and `EncryptedData`. Ok for you?

Reply:
@vsevel np at all :-), I'm always looking forward to some live discussions :-) and I'm finding them useful and educational. Right, `BInaryContent` does not sound as good to me now as it used to earlier. 'ClearData' was unusual first, but when I looked at the interface, I thought it was just perfect. `EncryptedData` is also good, `Data` parts are synced, and it is less technical than `Ciphertext`, now and then I check myself, what that really means, just an inout to the encryption or the result :-) so yeah, if you happy enough with the current names then lets keep them this way and of course that can be changed again before the merge if will be preferred in the end

Reply:
@vsevel This morning I have a slightly different view, `ClearData` is still good to me, but `EncryptedData`, I don't know, it is only and always a `String`. If we document that `encrypt` operations return an `encrypted data` and name the `decrypt` parameters as `encryptedData` then it will be clear enough to the users ? Yes we do have a wrapper for the clear data but it is really because we expect not only a String, and it is needed for the `decrypt` response

Reply:
done